### PR TITLE
Support vpm and resampler module of minicpm-v on NPU

### PR DESCRIPTION
**Pull request by plusbang**
## Description

Update minicpm-v usage on NPU.

### 2. User API changes

- No need to specify `torch_dtype=torch.float32` and `modules_to_not_convert=['vpm', 'resampler']`
- lm_head of minicpm_v_2_6 could run on NPU by default

### 3. Summary of the change 

- replace conv2d and layernorm with `MinicpmVPatchEmbedding` and `MinicpmVLayerNorm`
- pad mlp.fc2 and replace forward function to avoid compile error
- pad lm_head and replace forward function to avoid compile error
- port attention / multi-head-attention / resampler forward functions
- update example script

### 4. How to test?
- [x] Application test
https://github.com/analytics-zoo/nano/issues/1724#issuecomment-2467282958

### CODE REVIEW

File: python/llm/src/ipex_llm/transformers/npu_models/minicpmv_mp.py [hunk 1]
```diff
@@ -0,0 +1,540 @@
...
+
+        attn_output = self.out_proj(attn_output)
+        attn_output = attn_output.view(tgt_len, bsz, attn_output.size(1))
+        if not is_batched:
+            # squeeze the output if input was unbatched
+            attn_output = attn_output.squeeze(1)
+        return attn_output, None
+
+
+def resampler_forward(self, x, tgt_sizes=None):
```
Comment (rnwang04) [inline python/llm/src/ipex_llm/transformers/npu_models/minicpmv_mp.py:500]: maybe comment this function is ported from where.

File: python/llm/src/ipex_llm/transformers/npu_models/minicpmv_mp.py [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+    if b is None:
+        b_q = b_k = b_v = None
+    else:
+        b_q, b_k, b_v = b.chunk(3)
+    return linear(q.float(), w_q.float(), b_q.float()), \
+        linear(k.float(), w_k.float(), b_k.float()), \
+        linear(v.float(), w_v.float(), b_v.float())
+
+
+def multi_head_attn_forward(
```
Comment (rnwang04) [inline python/llm/src/ipex_llm/transformers/npu_models/minicpmv_mp.py:325]: maybe comment this function is ported from where.

**Review by rnwang04 — APPROVED** [review]
others LGTM

File: python/llm/src/ipex_llm/transformers/npu_models/minicpmv_mp.py (hunk 1 above)
Comment (plusbang) [reply python/llm/src/ipex_llm/transformers/npu_models/minicpmv_mp.py:501]: > maybe comment this function is ported from where.

Sure, have added related message : )

File: python/llm/src/ipex_llm/transformers/npu_models/minicpmv_mp.py [hunk 3]
```diff
@@ -0,0 +1,542 @@
...
+    hidden_states = self.fc2(hidden_states)
+    return hidden_states
+
+
+def pad_lm_head(module: torch.nn.Module):
+    if hasattr(module, 'lm_head') and module.lm_head.in_features == 3584 \
+       and module.lm_head.out_features == 151666:
+        new_linear = torch.nn.Linear(0, 0, bias=False)
+        padded_weight = F.pad(module.lm_head.weight,
+                              (0, 0, 0, 152064-151666))  # 152064 is qwen2-7b vocab_size
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/transformers/npu_models/minicpmv_mp.py:236]: Will this impact accuracy for channel-wise?

Comment (plusbang) [reply python/llm/src/ipex_llm/transformers/npu_models/minicpmv_mp.py:236]: > Will this impact accuracy for channel-wise?

The original weight shape is [151666, 3584] and pad it to [152064, 3584], each row has no change. I think this don't influence CW : )
